scipy/stats/morestats.py

```diff
@@ -1018,27 +1033,38 @@ def boxcox(x, lmbda=None, alpha=None, optimizer=None):
 
     >>> fig = plt.figure()
     >>> ax1 = fig.add_subplot(211)
-    >>> x = stats.loggamma.rvs(5, size=500) + 5
+    >>> x = stats.loggamma.rvs(5, size=500, random_state=0) + 5
     >>> prob = stats.probplot(x, dist=stats.norm, plot=ax1)
     >>> ax1.set_xlabel('')
     >>> ax1.set_title('Probplot against normal distribution')
 
     We now use `boxcox` to transform the data so it's closest to normal:
 
     >>> ax2 = fig.add_subplot(212)
-    >>> xt, _ = stats.boxcox(x)
+    >>> xt, lmbda = stats.boxcox(x)
     >>> prob = stats.probplot(xt, dist=stats.norm, plot=ax2)
     >>> ax2.set_title('Probplot after Box-Cox transformation')
 
     >>> plt.show()
 
-    Alternatively, we can define our own `optimizer` function.
+    The value of `lmbda` that maximizes the log-likelihood function is:
+    >>> lmbda
+    2.769...
+
+    Suppose we are only interested in values of `lmbda` on the interval [3, 4],
+    and we want to use `scipy.optimize.minimize_scalar` with
+    ``method='bounded'``to perform optimization of the log-likelihood function.
+    To do this, we define a custom `optimizer` function that accepts
+    positional argument `fun` and keyword argument `args` and passes them to
+    `scipy.optimize.minimize_scalar` with the desired options:
 
     >>> from scipy import optimize
     >>> def optimizer(fun, args):
-    ...     return optimize.minimize_scalar(fun, bounds=(0, 1), args=args,
+    ...     return optimize.minimize_scalar(fun, bounds=(3, 4), args=args,
     ...                                     method="bounded")
-    >>> xt, _ = stats.boxcox(x, optimizer=optimizer)
+    >>> xt, lmbda = stats.boxcox(x, optimizer=optimizer)
+    >>> lmbda
+    3.000...
```

Comment:
Actually, instead of including this whole example, what do you think about just referring the reader to `boxcox_normmax` for the example?

Reply:
Yes I agree, let's keep this in a single place.